safe/processors/pre_processors.py

```diff
@@ -79,3 +82,59 @@ def fake_nearby_preprocessor(impact_function):
         'value': layer_purpose_nearby_places,
     }
 }
+
+
+def check_earthquake_contour_preprocessor(impact_function):
+    """Checker for the contour preprocessor.
+
+    :param impact_function: Impact function to check.
+    :type impact_function: ImpactFunction
+
+    :return: If the preprocessor can run.
+    :rtype: bool
+    """
+    hazard_key = impact_function.hazard.keywords.get('hazard')
+    is_earthquake = hazard_key == hazard_earthquake['key']
+
+    geometry_key = impact_function.hazard.keywords.get('layer_geometry')
```

Comment:
this keyword is not very well maintained. Why not using the QGIS API to check this property? `is_raster_layer` exists in `safe/utilities/gis.py` See ticket https://github.com/inasafe/inasafe/issues/4519

Reply:
Done.